### DISCUSSION TITLE
Resource hook does not provide original attributes.

### DISCUSSION THREAD

**Discussion by stefan-pfeifer** [laravel/nova-issues#5633] [Q&A] [answered]
- Laravel Version: 10.10.0
- Nova Version: 4.24.2
- PHP Version: 8.2.0

#### Expectation: 
When the [resource hook](https://nova.laravel.com/docs/4.0/resources/#resource-hooks) `afterUpdate(NovaRequest $request, Model $model)` is called:
`$model->original` should contain the old values as they were before updating.  
`$model->attributes` should contain the updated values.

#### Reality: 
`$model->original` contains the updated values, not the original ones.

#### Workaround: 
Use a regular Laravel [observer](https://laravel.com/docs/10.x/eloquent#observers) instead of Nova hooks.

**Comment by crynobone** [comment, marked as answer]
`Resource::afterCreate()` and `Resource::afterUpdate()` is only fired after `$model->save()` and at this point original attributes has been synced. You should use Laravel Observer if you have such requirements.
